src/UI/Component/Signal.php

```diff
@@ -12,4 +12,18 @@
  */
 interface Signal {
 
+	/**
+	 * Get the ID of this signal
+	 *
+	 * @return string
+	 */
+	public function getId();
+
+	/**
+	 * Get the options of this signal
+	 *
+	 * @return array
+	 */
+	public function getOptions();
```

Comment:
I agree. How would I access these options in a renderer? Adding this method on the concrete implementation only seems wrong.